Add option for LUAUNDERLINE

briochemc opened the pull request:
I tried my best giving a shot at a PR that adds a `LUAUNDERLINE` option for latexdiff to use the lua-ul package (essentially what was suggested in #255 by @michaelmhoffman), which avoids the need for `mbox`es required by the ulem package. LuaLaTeX is then required to compile the differences. 

Checking for ulem was already baked in latexdiff so I didn't add anything for that.

I ran the tests mostly successfully, my use case being mostly for allowing line breaks in citations. See comparison below (taken from the apacite test):

| UNDERLINE (citations leak in the margin) | LUAUNDERLINE (nice line breaks!) | 
|---|---|
|  <img width=100% alt="Screenshot 2024-03-20 at 2 07 13 pm" src="https://github.com/ftilmann/latexdiff/assets/4486578/bd0c19a3-9035-4c1c-9520-47778d9047fc"> |  <img width=100% alt="Screenshot 2024-03-20 at 2 07 08 pm" src="https://github.com/ftilmann/latexdiff/assets/4486578/4a8fe815-a25c-4a40-abfa-420f29b587aa"> |

Potentially fixes #17, #60, #188, #255, #270, #277, and more?

I hope this helps! Let me know what else is needed!

PS: For tests I had to make some changes to some `usepackage` and my editor automatically removes trailing white spaces so sorry about that noise...

briochemc commented:
@ftilmann could you let me know if I need to change anything?

ftilmann commented:
Thanks a lot for your PR. I am sorry; I usually cannot be very quick in responding due to other demands on my time. I had a quick look in github. For "verify" I would like to keep pdflatex as the default for `--run` option. Would it be possible to adapt that so that processing with luatex becomes another option, e.g. `--luarun` [don't worry, I could also take care of that]. For latexdiff it was difficult to understand what the changes were with all the final space changes.  Not a problem but I need to go through command line to choose diff options that ignore that change. I hope to be able to manage tomorrow, but might be next weekend, or in worst case in two weeks. Sorry for that.

briochemc commented:
Thanks for your feedback and no worries! I'm sorry BTW, I didn't mean to push you, I completely understand that this amazing work of yours is done on your free time. I merely wanted to know how I could help more.

I will try to give a go at implementing these when I get the time myself :)

ftilmann commented:
OK, I had quick look.  Great to see that it works out of the box by providing simply a new style option. This make is easy to accept in principle though I could not actually verify that the compilations because my texlive installation on Ubuntu 20 is too old to include lua-ul (I know I could have installed by hand but as you made these tests already I did not really see the need).

Some minor issues:
- I noticed one typo in the man page markup: in line  4245 of modified latexdiff it should be LUAUNDERLINE (instead of UNDERLINE) I think.
- As from one perspective only another markup style has been defined I would prefer to leave the version number 1.3.4a instead of 1.4.0 (I will make a release soon, which would contain this additional markup (lines 36,277)
- The modified .tex files in testsuit, e.g. tested with gershwin12-new.tex no longer compiled for me after changing to aguplus class. I have installed aguplus.cls via the repo texlive-publishers on Ubuntu20, and I get the error `! Class aguplus Error: No valid AGUTeX package given as option.`  [I made sure my own aguplus.cls files does not shadow).  I did not investigate further.  I admit that relying on obsolete agu2001.cls file is also not great, but at least for me that runs through. What version of aguplus do you  have exactly?   For now it would be great if you could revert these changes.

briochemc commented:
Sorry for the delay, I was on holiday! Anyway, in the latest commit I have addressed your 3 points:
- [x] Fixed two typos where it should have been `LUAUNDERLINE` (the one you found and another one)
- [x] Reverted the version number (in the lines you mentioned and a third one)
- [x] Reverted all the test changes 

I hope this helps! (Again, best to view the changes without trailing space noise with https://github.com/ftilmann/latexdiff/pull/299/files?w=1)

briochemc commented:
Gentle ping @ftilmann in case you can spare some time to look into merging this :)

ftilmann commented:
Very sorry about the delay - I merged without any further modifications. Thanks a lot for contributing this commit.

briochemc commented:
No worries! Glad I could contribute something useful!

briochemc commented:
I just noticed that the CTAN version has not been updated since the last version (1.3.3, dating back from 2022). Is it possible to release 1.3.4 on CTAN as well? Sorry for noise in case this delay is on purpose!

ftilmann commented:
Thanks for the reminder. Indeed that task got lost over the summer holidays.  I just uploaded to 1.3.4 to CTAN - usually takes a little while until it is checked by moderators, but should be available through CTAN in a few days at most.
